app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, optional: true, foreign_key: :users_id
+  has_many :likes, foreign_key: :posts_id
+  has_many :comments, foreign_key: :posts_id
+
```

Comment:
**OPTIONAL**
- Also, please consider using the `after_save` method to call the `post_counter` every time the model changes!

_Please consider these optional changes in all of your models!_ 🙏